src/components/structures/MessagePanel.tsx

```diff
@@ -641,20 +641,21 @@ export default class MessagePanel extends React.Component<IProps, IState> {
         // to assume that sent receipts are to be shown more often.
         this.readReceiptsByEvent = {};
         if (this.props.showReadReceipts) {
-            this.readReceiptsByEvent = this.getReadReceiptsByShownEvent();
+            this.readReceiptsByEvent = this.getReadReceiptsByShownEvent(events);
         }
 
         let grouper: BaseGrouper | null = null;
 
         for (let i = 0; i < events.length; i++) {
-            const { event: mxEv, shouldShow } = events[i];
-            const eventId = mxEv.getId();
-            const last = mxEv === lastShownEvent;
+            const ev = events[i];
```

Comment:
I'm slightly struggling to keep in my head which things are `MatrixEvent`s and which `EventAndShouldShow`s. Rename to `eventAndShouldShow`?

Reply:
Done